src/app/auth/auth.service.ts

```diff
@@ -13,37 +13,35 @@ export class AuthServices{
     toastr = inject(ToastrService);
     storage = inject(Storage);
 
-    register(email: string, username: string, password: string): Observable<{ success: boolean }>{
-        const result = createUserWithEmailAndPassword(
-            this.firebaseAuth,
-            email,
-            password,
-        ).then(async (result) => {
+    public async register(email: string, username: string, password: string): Promise<{ success: boolean }> {
```

Comment:
This is much cleaner now! 😊 However, I noticed that when a user successfully registers, you return a status instead of the user information. Shouldn't the return type be status rather than success? Because when an error occurs, it's not considered a success but rather the opposite